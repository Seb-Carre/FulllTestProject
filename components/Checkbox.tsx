import {Pressable, StyleSheet, View} from 'react-native';

import {Icon} from './Icon';
import React from 'react';

type CheckboxParams = {
  mode: 'active' | 'multiple' | 'none';
  onPress: () => void;
  testID?: string;
};

/**
 * Functional Component to return a checkbox
 * The check mark is the only thing it isn't made with style
 */
export default function Checkbox({mode, onPress, testID}: CheckboxParams) {
  // CheckMark icon to show when active mode is enabled
  const checkMark = '../assets/images/check-mark.png';
  return (
    <Pressable
      onPress={onPress}
      testID={testID}
      style={[
        styles.container,
        mode === 'active' || mode === 'multiple'
          ? styles.containerActive
          : null,
      ]}>
      {/* We are going to take the corresponding style we made in stylesheet styles, active mode is the only one mode which not made with styling */}
      {mode !== 'active' && (
        <View testID="Checkbox.viewStyle" style={styles[mode]} />
      )}
      {/* if active, the need to use the icon asset image to show it, this is the only mode which is different */}
      {mode === 'active' && (
        <Icon
          testID="Checkbox.icon"
          name={require(checkMark)}
          style={styles.active}
        />
      )}
    </Pressable>
  );
}

const styles = StyleSheet.create({
  container: {
    height: 20,
    width: 20,
    borderWidth: 4,
    borderRadius: 4,
    justifyContent: 'center',
  },
  containerActive: {backgroundColor: '#44a1f8'},
  multiple: {height: 3, backgroundColor: 'black'},
  active: {
    height: 13,
    width: 13,
  },
  none: {},
});
