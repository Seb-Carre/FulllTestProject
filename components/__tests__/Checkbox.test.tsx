import {fireEvent, render} from '@testing-library/react-native';

import Checkbox from '../Checkbox';
import React from 'react';

it('render the checkbox with active mode as snapShot we expected', () => {
  const {toJSON} = render(<Checkbox onPress={() => {}} mode="active" />);

  expect(toJSON()).toMatchSnapshot();
});

it('check if function is being called by pressing the checkbox', () => {
  // Using a mock function to check if our function is called
  const functionToCall = jest.fn();
  const {getByTestId} = render(
    <Checkbox onPress={functionToCall} testID="checkbox" mode="active" />,
  );
  // Making a press event on our checkbox
  fireEvent.press(getByTestId('checkbox'));
  expect(functionToCall).toBeCalled();
});

it('check if icon is here with active mode', () => {
  const {queryByTestId} = render(<Checkbox onPress={() => {}} mode="active" />);
  // if the mode is active, we have in this case an icon component !
  expect(queryByTestId('Checkbox.icon')).not.toBeNull();
});

it('check if view style is visible with multiple mode', () => {
  const {queryByTestId} = render(<Checkbox onPress={() => {}} mode="multiple" />);
  // if the mode is multiple, we have in this case our viewStyle component !
  expect(queryByTestId('Checkbox.viewStyle')).not.toBeNull();
});

it('check if view style is visible with none mode', () => {
  const {queryByTestId} = render(<Checkbox onPress={() => {}} mode="none" />);
  expect(queryByTestId('Checkbox.viewStyle')).not.toBeNull();
});
