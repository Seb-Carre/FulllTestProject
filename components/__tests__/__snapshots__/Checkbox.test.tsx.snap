// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the checkbox with active mode as snapShot we expected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#44a1f8",
      "borderRadius": 4,
      "borderWidth": 4,
      "height": 20,
      "justifyContent": "center",
      "width": 20,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/images/check-mark.png",
      }
    }
    style={
      {
        "height": 13,
        "width": 13,
      }
    }
    testID="Checkbox.icon"
  />
</View>
`;
